src/main/java/xbot/common/controls/actuators/mock_adapters/MockCANMotorController.java

```diff
@@ -0,0 +1,95 @@
+package xbot.common.controls.actuators.mock_adapters;
+
+import dagger.assisted.Assisted;
+import dagger.assisted.AssistedFactory;
+import dagger.assisted.AssistedInject;
+import edu.wpi.first.units.measure.Angle;
+import edu.wpi.first.units.measure.AngularVelocity;
+import edu.wpi.first.units.measure.Time;
+import xbot.common.controls.actuators.XCANMotorController;
+import xbot.common.injection.DevicePolice;
+import xbot.common.injection.electrical_contract.CANMotorControllerInfo;
+import xbot.common.injection.electrical_contract.CANMotorControllerOutputConfig;
+import xbot.common.properties.PropertyFactory;
+
+import static edu.wpi.first.units.Units.Degrees;
+import static edu.wpi.first.units.Units.RPM;
+
+public class MockCANMotorController extends XCANMotorController {
+    @AssistedFactory
+    public abstract static class MockCANMotorControllerFactory implements XCANMotorControllerFactory {
+        public abstract MockCANMotorController create(
+                @Assisted("info") CANMotorControllerInfo info,
+                @Assisted("owningSystemPrefix") String owningSystemPrefix,
+                @Assisted("pidPropertyPrefix") String pidPropertyPrefix);
+    }
+
+    @AssistedInject
+    public MockCANMotorController(
+            @Assisted("info") CANMotorControllerInfo info,
+            @Assisted("owningSystemPrefix") String owningSystemPrefix,
+            PropertyFactory propertyFactory,
+            DevicePolice police,
+            @Assisted("pidPropertyPrefix") String pidPropertyPrefix
+    ) {
+        super(info, owningSystemPrefix, propertyFactory, police, pidPropertyPrefix);
+    }
+
+    @Override
+    public void setConfiguration(CANMotorControllerOutputConfig outputConfig) {
+
+    }
+
+    @Override
+    public void setOpenLoopRampRates(Time dutyCyclePeriod, Time voltagePeriod) {
+
+    }
+
+    @Override
+    public void setClosedLoopRampRates(Time dutyCyclePeriod, Time voltagePeriod) {
+
+    }
+
+    @Override
+    public void setPidProperties(double p, double i, double d, double velocityFF, int slot) {
+
+    }
+
+    @Override
+    public void setPower(double power) {
```

Comment:
⭐ something we can do later, but it would be nice if these stored the last things set so it could be read back out by unit tests/physics sims